### PR TITLE
104 -- Added ACS5 2015 fields for "good neighborhood" variables

### PR DESCRIPTION
**Pull request by MikeS-nth**
@NealHumphrey I updated census.py with the fields that can be used to calculate the "good neighborhood" variables from the Harvard study.  (line 67)

I tried to run it from the command line, but it threw an error related to base.py.  I think it may be a Python 2.7 vs 3.x thing, but I can figure that out later.

### CODE REVIEW

**Comment by MikeS-nth** [comment]
I just pushed a fix to the code.  The last field name was a dupe, I replaced it with the correct one.

File: .spyproject/codestyle.ini [hunk 1]
```diff
@@ -0,0 +1,6 @@
+[codestyle]
+indentation = True
+
+[main]
+version = '0.1.0'
+
```
Comment (ajhalani) [inline .spyproject/codestyle.ini]: Is the committing of these .spyproject/* files intentional?

File: python/housinginsights/sources/census.py [hunk 2]
```diff
@@ -64,38 +64,38 @@ def __init__(self, proxies=None):
                         "acs5_2012"     : "NAME,B01003_001E,B25057_001E,B25058_001E,B25059_001E",
                         "acs5_2013"     : "NAME,B01003_001E,B25057_001E,B25058_001E,B25059_001E",
                         "acs5_2014"     : "NAME,B01003_001E,B25057_001E,B25058_001E,B25059_001E",
-                        "acs5_2015"     : "NAME,B01003_001E,B25057_001E,B25058_001E,B25059_001E",
+                        "acs5_2015"     : "NAME,B01003_001E,B25057_001E,B25058_001E,B25059_001E, B02001_003E, B17020_002E, B19025_001E, B23025_002E, B16008_019E, B09002_015E",
```
Comment (ajhalani) [inline python/housinginsights/sources/census.py]: Ideally, field names should be consistent in all the years and margin of error.. Is their a specific reason why change is limited to acs5_2015?

File: docs/data/census.md [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+|:---|:---|:---|
+|[B02001_003E](http://api.census.gov/data/2015/acs5/variables/B02001_003E.json) | Black or African American alone | B02001. Race |
+|[B17020_002E](http://api.census.gov/data/2015/acs5/variables/B17020_002E.json) |Income in the past 12 months below poverty level |B17020. Poverty Status in the Past 12 Months by Age|
+|[B19025_001E](http://api.census.gov/data/2015/acs5/variables/B19025_001E.json) |Aggregate household income in the past 12 months (in 2015 Inflation-adjusted dollars) |B19025. Aggregate Household Income in the Past 12 Months (in 2015 Inflation-Adjusted Dollars) |
+|[B23025_002E](http://api.census.gov/data/2015/acs5/variables/B23025_002E.json) |In labor force |B23025. Employment Status for the Population 16 Years and Over |
+|[B16008_019E](http://api.census.gov/data/2015/acs5/variables/B16008_019E.json) |Foreign-born population |B16008. Citizenship Status by Age by Language Spoken At Home and Ability to Speak English for the Population 5+ Yrs |
+|[B09002_015E](http://api.census.gov/data/2015/acs5/variables/B09002_015E.json) |In other families:!!Female householder, no husband present |B09002. OWN CHILDREN UNDER 18 YEARS BY FAMILY TYPE AND AGE |
```
Comment (ajhalani) [inline docs/data/census.md]: Could we extend this to existing fields in census.py as well ?

Comment (MikeS-nth) [inline docs/data/census.md]: Definitely.  Was focused for now on the fields I was looking at related to issue 104.

(I actually didn't mean for this file to be in this PR.  I have a separate branch for it where I'm continuing to edit and will create a different PR.  Sorry for my git clumsiness; still getting up to speed on this project.)

File: .spyproject/codestyle.ini (hunk 1 above)
Comment (MikeS-nth) [inline .spyproject/codestyle.ini]: I'm not sure where those came from.  I think they got swept up in the `git add -A`

File: python/housinginsights/sources/census.py (hunk 2 above)
Comment (MikeS-nth) [inline python/housinginsights/sources/census.py]: No, didn't have a specific reason.  I think I assumed (maybe prematurely) that the "good neighborhood" specific fields would be used differently and/or the most recent year would be the most relevant.

I can easily add the rest in, including the related margin of error fields.

**Comment by MikeS-nth** [comment]
Pushed an updated file with the fields filled in for 2009 - 2014, as well as all matching margin of error fields.  
Note, there were some fields that were not consistent across all years:
- For years 2009 - 2012, the fields B17020_002E & B17020_002M (income below poverty level) did not exist; I found a suitable replacement with B17001_002E & B17001_002M for those years
- For years 2009 and 2010, the fields B23025_002E and B23025_002M (number of people in the labor force) did not exist.  I have not yet found a suitable replacement.

File: docs/data/census.md (hunk 3 above)
Comment (NealHumphrey) [inline docs/data/census.md]: @MikeS-nth  -in this pull request (for adding the fields to python code), I can manually remove this file completely before merging. But for your other pull request, see my notes about how to fix up the pull request to get rid of unwanted files, as well as a checklist you can refer to for git branching strategies to keep this from happening.

**Comment by NealHumphrey** [comment]
Great, thanks for the update. 

for income below poverty level, are those fields the exact same thing? i.e. can we put the data into the same column in our database and use it in the same way in all cases?

Otherwise looks good, I'll fix the extra file in this PR (but let you fix the one in the documentation pr). Thanks @ajhalani for your review as well!

**Comment by MikeS-nth** [comment]
@NealHumphrey 
Based on my reading of the descriptions, I believe they are measuring the same thing, but are organized somewhat differently.  The 2013-2015 fields are in a "by Age" category, and the 2009 - 2012 fields are in the "by Sex by Age" category.  Despite that difference, the fields appear to be the category totals, and should therefore be describing the same sample (since all respondents logically  must have a sex and age, there *should* be perfect overlap unless there was missing data or sampling error).  

Does that make sense to you?  I don't want to overstate my confidence, as I'm new to working with census data.


Here are the descriptions from the census field list pages:

[B17020_002E](http://api.census.gov/data/2015/acs5/variables/B17020_002E.json): 
- 2013 - 2015
- Name: Income in the past 12 months below poverty level
- Concept: B17020. Poverty Status in the Past 12 Months by Age
- ([B17020_002M](http://api.census.gov/data/2015/acs5/variables/B17020_002M.json) is margin of error)
- [field list link](http://api.census.gov/data/2015/acs5/variables.html)

[B17001_002E](http://api.census.gov/data/2009/acs5/variables/B17001_002E.json)
- 2009 - 2012
- Name: Income in the past 12 months below poverty level
- Concept: B17001. Poverty Status in the past 12 Months by Sex by Age
- ([B17001_002M](http://api.census.gov/data/2009/acs5/variables/B17001_002M.json) is margin of error)
- [field list link](http://api.census.gov/data/2009/acs5/variables.html)

**Comment by NealHumphrey** [comment]
@MikeS-nth I think that makes sense with the replacement field. If you can briefly summarize that in the update to the census.md documentation file that would be great!

**Comment by NealHumphrey** [comment]
Merged on the command line, not sure why this isn't showing up as merged here. Closing as it's in the current code.

**Pull request closed without merging**